File: Go/Easy/Selection Sort/solution_1.go
package main

func SelectionSort(array []int) []int {

	for i := 0; i < len(array); i++ {

		k := i

		for j := i + 1; j < len(array); j++ {

			if array[k] > array[j] {
				k = j
			}
		}
		swap(k, i, array)
	}
	return array
}

func swap(index int, indexTwo int, array []int) {
	array[index], array[indexTwo] = array[indexTwo], array[index]
}

File: Python/Easy/Selection Sort/solution_1.py

def selectionSort(array):
   
    for i in range(len(array)):
        k = i
        for j in range(i + 1, len(array)):
            if array[k] > array[j]:
                k = j
        swap(k, i, array)

    return array

def swap(index, indexTwo, array):
    array[index],array[indexTwo] = array[indexTwo] , array[index]